cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java

```diff
@@ -207,12 +242,13 @@ private AuthenticationClient getAuthenticationClient(ConnectionConfig connection
     return authenticationClient;
   }
 
-  private AccessToken getSavedAccessToken(String hostname) {
+  private UserAccessToken getSavedAccessToken(String hostname) {
     File file = getAccessTokenFile(hostname);
     if (file.exists() && file.canRead()) {
       try {
-        String tokenString = Joiner.on("").join(Files.readLines(file, Charsets.UTF_8));
-        return new AccessToken(tokenString, -1L, null);
+        FileReader reader = new FileReader(file);
```

Comment:
The reader is not closed after used. Also, always provides a Charset when creating `Reader`.

```java
try (Reader reader = Files.newReader(file, Charsets.UTF_8)) {
  return GSON.fromJson(reader, UserAccessToken.class);
}
```